target/linux/ipq806x/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac

```diff
@@ -29,3 +29,14 @@ for i in $(seq 0 9);
       fi
   fi
 done
+
+board=$(board_name)
+
+case "$board" in
+	ubnt,unifi-ap-hd)
+		[ "$PHYNBR" -eq 0 ] && \
+			macaddr_add "$(mtd_get_mac_binary EEPROM 0x6)" 1 > /sys${DEVPATH}/macaddress
+		[ "$PHYNBR" -eq 1 ] && \
+			macaddr_add "$(mtd_get_mac_binary EEPROM 0x6)" 2 > /sys${DEVPATH}/macaddress
```

Comment:
This also has no effect. Any obvious mistakes here?

Reply:
there could be something wrong with the script since it seems you are the first to use it

As a test, try without all the conditionals and variables...

but first make sure you can return the MAC properly with the functions
in openwrt shell:

```
. /lib/functions.sh
. /lib/functions/system.sh
macaddr_add "$(mtd_get_mac_binary EEPROM 0x6)" 1
```

then have these lines in the case of the script (without the $PHYNBR lines) if that works

```
		macaddr_add "$(mtd_get_mac_binary EEPROM 0x6)" 1 > /sys/class/ieee80211/phy0/macaddress
		macaddr_add "$(mtd_get_mac_binary EEPROM 0x6)" 2 > /sys/class/ieee80211/phy1/macaddress
```

and verify after flashing

`cat /sys/class/ieee80211/phy0/macaddress`

Reply:
also the cases must end with a catch-all case

```
	*)
		;;
esac
```

Reply:
looking at previous commits....this script used to have a list of cases but they were removed.

`echo` was used for each line, maybe that is the missing piece

see the script at this point in time, and put the case list before `OPATH=`

https://github.com/openwrt/openwrt/blob/38167cb7f41b316fcc6d6bbcab993a2f42f30c0e/target/linux/ipq806x/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac

Reply:
Thanks, I updated the script and pushed the changes. This now works, as in: Initial `ip addr` after boot shows wrong MAC addresses, but `/sys/class/ieee80211/pyhX/macaddress` shows the correct value. Once I setup any network on it, `ip addr` also shows the correct MAC address.

Is this expected behavior when patching the addresses through the script or is there some reload missing somewhere?

Reply:
I don't know but apparently, mac address moving from /sys to userspace relies on the driver being loaded or reloaded in some way
according to past commits, they had an issue where the phy number would also change on driver reload...

only problem now is whitespace for the wildcard `*)` case probably because you copy and pasted 👍🏽

Reply:
Well, to alter the MAC address on load of the interface, the interface needs to be loaded. That means if no interface is up because WiFi is disabled by default, the MAC won't be changed. This happens when the interface is loaded for the first time, as you observed.
This is one of the reasons why 10_fix_wifi_mac is the least desirable of all ways to adjust WiFi MAC address.